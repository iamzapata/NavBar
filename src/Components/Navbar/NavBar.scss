@import "styles/variables.scss";

.NavBar {
  button.navbar-burger.burger.is-active {
    border: 0;
    box-shadow: none;
    background-color: transparent !important;
  }
  padding: 0 20px;

  .navbar-bra {
    min-width: 180px;
  }

  .navbar-start {
    margin: 0 auto;
    justify-content: center;
  }

  .navbar-end {
    margin: 0;
    max-width: 180px;
  }

  a.navbar-item {
    color: $color-gray-text;

    &:hover {
      background: none;
      color: $color-black-text;
    }

    &.NavLink_Selected {
      color: $color-black-text;
    }
  }
}

@media only screen and (max-width: 769px) {
  .NavBar {
    button.navbar-burger.burger {
      background-color: transparent;
      border: 0;
    }
    .navbar-end {
      width: 100%;
      max-width: inherit;
      display: flex;
      align-items: center;
      justify-content: space-between;
    }
  }
}
