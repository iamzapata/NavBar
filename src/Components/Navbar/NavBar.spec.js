import React from "react"
import { render, cleanup, fireEvent } from "react-testing-library"
import configureMockStore from "redux-mock-store"
import { Provider } from "react-redux"
import { BrowserRouter as Router } from "react-router-dom"
import NavBar from "./index"
import "jest-dom/extend-expect"

const mockStore = configureMockStore()
const store = mockStore({ Theme: { themeName: "Pinterest" } })

describe("<NavBar />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>
    )
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("Should toggle burger menu", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>
    )
    const burgerButton = container.querySelector("button")
    const navbarMenu = container.querySelector(".navbar-menu")

    fireEvent.click(burgerButton)
    expect(burgerButton).toHaveClass("is-active")
    expect(burgerButton).toHaveClass("is-active")

    fireEvent.click(burgerButton)
    expect(burgerButton).not.toHaveClass("is-active")
    expect(navbarMenu).not.toHaveClass("is-active")
  })
})
