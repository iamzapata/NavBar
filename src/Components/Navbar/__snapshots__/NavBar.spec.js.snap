// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Should render self 1`] = `
<nav
  class="NavBar navbar"
>
  <div
    class="navbar-brand"
  >
    <a
      aria-current="page"
      class="NavLInkBrand navbar-item active"
      href="/"
    >
      <img
        alt="Pinterest"
        src="test-file-stub"
      />
      <span
        class="BrandName is-size-5"
      >
        Pinterest
      </span>
    </a>
    <button
      class="navbar-burger burger"
      type="button"
    >
      <span />
      <span />
      <span />
    </button>
  </div>
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
      role="button"
      tabindex="0"
    >
      <a
        class="navbar-item"
        href="/feed"
      >
        News
      </a>
      <a
        class="navbar-item"
        href="/people"
      >
        People
      </a>
      <a
        class="navbar-item"
        href="/food"
      >
        Food
      </a>
      <a
        class="navbar-item"
        href="/admin"
      >
        Admin
      </a>
      <a
        class="navbar-item"
        href="/more"
      >
        More
      </a>
      <div
        class="SearchBar control has-icons-left navbar-item"
      >
        <input
          class="input"
          placeholder="Find a Pintrest"
          type="text"
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fa fa fa-search"
          />
        </span>
      </div>
    </div>
    <div
      class="navbar-end"
    >
      <div
        class="NotificationsTray navbar-item"
      >
        <button
          class="button"
          type="button"
        >
          <i
            class="fa fa-bell"
          />
          <span
            class="Notifications__Count"
          >
            7
            +
          </span>
        </button>
      </div>
      <div
        class="ProfileDropdown dropdown navbar-item"
      >
        <div
          class="dropdown-trigger"
        >
          <button
            aria-controls="ProfileDropdown"
            aria-haspopup="true"
            class="button"
            data-testid="ProfileButton"
            type="button"
          >
            <img
              alt="Profile"
              class="image"
              src="test-file-stub"
            />
            <span
              class="icon is-small"
            >
              <i
                aria-hidden="true"
                class="fa fa-caret-down"
              />
            </span>
          </button>
        </div>
        <div
          class="dropdown-menu"
          id="ProfileDropdown"
          role="menu"
        >
          <div
            class="dropdown-content"
            role="button"
            tabindex="0"
          >
            <a
              class="dropdown-item"
              href="/profile/1"
            >
              Profile
            </a>
            <a
              class="dropdown-item"
              href="/settings"
            >
              Settings
            </a>
            <a
              class="dropdown-item"
              href="/sign-out"
            >
              Sign Out
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
